src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java

```diff
@@ -192,60 +252,125 @@ public WindowBuildTool(@Nullable final BlockPos pos)
         registerButton(BUTTON_UP, WindowBuildTool::moveUpClicked);
         registerButton(BUTTON_DOWN, WindowBuildTool::moveDownClicked);
         registerButton(BUTTON_ROTATE_RIGHT, this::rotateRightClicked);
         registerButton(BUTTON_ROTATE_LEFT, this::rotateLeftClicked);
-        registerButton(BUTTON_MIRROR, WindowBuildTool::mirror);
 
+        registerButton(BUTTON_RENAME, this::renameClicked);
+        registerButton(BUTTON_DELETE, this::deleteClicked);
+        renameButton = findPaneOfTypeByID(BUTTON_RENAME, Button.class);
+        deleteButton = findPaneOfTypeByID(BUTTON_DELETE, Button.class);
     }
 
     /**
-     * Check if the player inventory has a certain hut.
-     *
-     * @param inventory the player inventory.
-     * @param hut       the hut.
-     * @return true if so.
+     * Initialise the previous/next and drop down list for section.
      */
-    private static boolean inventoryHasHut(@NotNull final InventoryPlayer inventory, final String hut)
+    private void initBuildingTypeNavigation()
     {
-        return inventory.hasItemStack(new ItemStack(Block.getBlockFromName(Constants.MOD_ID + HUT_PREFIX + hut)));
+        registerButton(BUTTON_PREVIOUS_TYPE_ID, this::previousSection);
+        registerButton(BUTTON_NEXT_TYPE_ID, this::nextSection);
+        sectionsDropDownList = findPaneOfTypeByID(DROPDOWN_TYPE_ID, DropDownList.class);
+        sectionsDropDownList.setHandler((c,i) -> onDropDownListChanged(c,i));
+        sectionsDropDownList.setDataProvider(new DropDownList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return sections.size();
+            }
+
+            @Override
+            public String getLabel(final int index)
+            {
+                final String name = sections.get(index);
+                if (Structures.SCHEMATICS_SCAN.equals(name))
+                {
+                    return LanguageHandler.format("com.minecolonies.coremod.gui.buildtool.scans");
+                }
+                else if (Structures.SCHEMATICS_PREFIX.equals(name))
+                {
+                    return LanguageHandler.format("com.minecolonies.coremod.gui.buildtool.decorations");
+                }
+                //should be a hut
+                return LanguageHandler.format("tile.minecolonies.blockHut" + name + ".name");
+            }
+        });
     }
 
     /**
-     * Move the schmatic up.
+     * Initialise the previous/next and drop down list for style.
      */
-    private static void moveUpClicked()
+    private void initStyleNavigation()
     {
-        Settings.instance.moveTo(new BlockPos(0, 1, 0));
+        registerButton(BUTTON_PREVIOUS_STYLE_ID, this::previousStyle);
+        registerButton(BUTTON_NEXT_STYLE_ID, this::nextStyle);
+        stylesDropDownList = findPaneOfTypeByID(DROPDOWN_STYLE_ID, DropDownList.class);
+        stylesDropDownList.setHandler((c,i) -> onDropDownListChanged(c,i));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1612)